test/metabase/analytics/prometheus_test.clj

```diff
@@ -152,3 +154,27 @@
                                    (metric-lines port))]
         (is (seq (set/intersection expected-lines actual-lines))
             "Registry does not have c3p0 metrics in it")))))
+
+(deftest email-collector-test
+  (testing "Registry has email metrics registered"
+    (with-prometheus-system [_ system]
```

Comment:
these tests are strange to me.
- they use interop instead of the library we are using.
```clojure
(some-> (get (:registry system) :metabase-email/messages)
        iapetos.operations/read-value)
0.0
```
- I would expect to see some behavior type tests: grab the value of the messages, send an email, check that the counter has increased. Then send an email that will trigger an error and check that the error counter has incremented.

Reply:
> * they use interop instead of the library we are using.
> 
> ```clojure
> (some-> (get (:registry system) :metabase-email/messages)
>         iapetos.operations/read-value)
> 0.0
> ```

Thanks! I replaced the Java interop with iapetos functions in https://github.com/metabase/metabase/pull/32467.

> * I would expect to see some behavior type tests: grab the value of the messages, send an email, check that the counter has increased. Then send an email that will trigger an error and check that the error counter has incremented.

I have such tests in `metabase.email-test/send-message!-test`.

That seemed to be the right place to test this, because (a) we already have the test setup available there, but more importantly because (b) `metabase.email` is the place where the actual counter increment happens, so it made sense to me to keep the test close to the code.